superset-frontend/src/views/components/RightMenu.tsx

```diff
@@ -296,10 +293,6 @@ const RightMenu = ({
     setShowDatabaseModal(false);
   };
 
-  const handleOnHideDatasetModalModal = () => {
```

Comment:
I think this is another situation where previously users were supposed to be able to create a dataset in the middle of doing something else – if we could maintain that behavior it would be awesome.  I just checked though and I think the current implementation is broken: it seem to always redirects to the add chart page no matter what 🤪

Reply:
The plans are to remove the old modal entirely, what other flow is this used in? Definitely could have missed some spots but I'd need to discuss with design since we won't be using the old modal anymore.

Reply:
Resolving because it will always redirect to Add Chart in this PR.